src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -1063,13 +1092,70 @@ void PerformInteractionRequest::SendBothModeResponse(
       msg_param.empty() ? NULL : &msg_param;
   std::string info = app_mngr::commands::MergeInfos(
       ui_perform_info, ui_info_, vr_perform_info, vr_info_);
+
   DisablePerformInteraction();
+
   SendResponse(result,
                perform_interaction_result_code,
                info.empty() ? NULL : info.c_str(),
                response_params);
 }
 
+mobile_apis::Result::eType
+PerformInteractionRequest::PrepareResultCodeForResponse(
+    const app_mngr::commands::ResponseInfo& ui_response,
+    const app_mngr::commands::ResponseInfo& vr_response) {
+  if (interaction_mode_ == mobile_apis::InteractionMode::VR_ONLY) {
+    if (first_responser_ == FirstPerformInteractionResponser::VR) {
+      return MessageHelper::HMIToMobileResult(vr_result_code_);
+    }
+  }
+
+  if (interaction_mode_ == mobile_apis::InteractionMode::BOTH) {
+    if (IsVRPerformInteractionResponseSuccessfulInBothMode()) {
+      return MessageHelper::HMIToMobileResult(vr_result_code_);
+    } else {
```

Comment:
@YarikMamykin `else` is redundant

Reply:
@AByzhynar fixed in https://github.com/smartdevicelink/sdl_core/pull/3122/commits/0190836450ab1d7e130bbb8e32315eb3325a9a68